src/transformers/dynamic_module_utils.py

```diff
@@ -614,6 +615,10 @@ def resolve_trust_remote_code(trust_remote_code, model_name, has_local_code, has
                     f"load the model. You can inspect the repository content at https://hf.co/{model_name}.\n"
                     f"Please pass the argument `trust_remote_code=True` to allow custom code to be run."
                 )
+            finally:
+                if prev_sig_handler is not None:
+                    signal.signal(signal.SIGALRM, prev_sig_handler)
```

Comment:
this will restore to the signal before the `try` not to the signal before `load_custom_code`. Should we not store `signal.getsignal(signal.SIGALRM)` before all that?

Reply:
Sorry, I didn't find where `load_custom_code` is, but I don't think we should reset it here.

Reply:
Oh no my bad. LGTM